source/validations.ts

```diff
@@ -0,0 +1,57 @@
+import type { Request } from 'express'
+import { ValidationLevel } from './types.ts'
+
+function runCheck(
+	level: ValidationLevel,
+	check: () => boolean,
+	errorMessage: string,
+) {
+	if (level === ValidationLevel.Off) {
+		return
+	}
+
+	if (!check()) {
+		const message = `express-rate-limit: ${errorMessage}`
+		if (level === ValidationLevel.Warn) {
+			console.warn(message)
+		} else if (level === ValidationLevel.Throw) {
+			throw new Error(message)
+		}
+	}
+}
+
+// https://www.regextester.com/104038
+// matches IPv4 and IPv6, but not if they include a port number at the end
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const reIP =
```

Comment:
I should probably just import an npm module to do this.. but I kind of like this being a zero-dependency library.